portfolio/src/main/webapp/api-script/script.js

```diff
@@ -0,0 +1,87 @@
+mapOptions = {
+  center: { lat: 0, lng: 0 },
+  zoom: 3,
+  scaleControl: false,
+  streetViewControl: false,
+  mapTypeControlOptions: {
+    mapTypeIds: ["moon"]
+  }
+};
+
+
+async function loadMap() {  
+  const response = await fetch("/landing-data");
+  const payload = await response.json();
+
+  
+  var map = new google.maps.Map(document.getElementById("map"), mapOptions);
+
+  var moonMapType = new google.maps.ImageMapType({
+    getTileUrl: function (coord, zoom) {
+      var normalizedCoord = getNormalizedCoord(coord, zoom);
+      if (!normalizedCoord) {
+        return null;
+      }
+      var bound = Math.pow(2, zoom);
+      return (
+        "https://mw1.google.com/mw-planetary/lunar/lunarmaps_v1/apollo" +
+        "/" +
+        zoom +
+        "/" +
+        normalizedCoord.x +
+        "/" +
+        (bound - normalizedCoord.y - 1) +
+        ".jpg"
+      );
+    },
+    tileSize: new google.maps.Size(256, 256),
+    maxZoom: 19,
+    minZoom: 0,
+    radius: 1738000,
+    name: "Moon"
+  });
+
+  map.mapTypes.set("moon", moonMapType);
+  map.setMapTypeId("moon");
+
+  var infowindow = new google.maps.InfoWindow();
+
+  var marker;
+
+  for (i in payload) {
+    const info = payload[i];
+    console.log(info);
+    const imgRef = "http://mw1.google.com/mw-planetary/lunar/lunarmaps_v1/util/s".concat(info.missionNum).concat(".png");
+    marker = new google.maps.Marker({ position: { lat: info.lat, lng: info.lng }, icon: imgRef, map: map });
+
+    google.maps.event.addListener(marker, 'click', (function (marker, i) {
+      return function () {
+        infowindow.setContent(info.description);
+        infowindow.open(map, marker);
+      }
+    })(marker, i));
+  }
+}
+
+// Normalizes the coords that tiles repeat across the x axis (horizontally)
+// like the standard Google map tiles.
+function getNormalizedCoord(coord, zoom) {
+  var y = coord.y;
+  var x = coord.x;
+
+  // tile range in one direction range is dependent on zoom level
+  // 0 = 1 tile, 1 = 2 tiles, 2 = 4 tiles, 3 = 8 tiles, etc
+  var tileRange = 1 << zoom;
```

Comment:
I'm personally of the opinion that bit shifting should only be allowed in C or assembly, and everything else should use multiplication or division :) Mostly because you don't want to force people to brush up on their bitshifting to understand your code.

But that's just me, others might be ok with it. So as long as you don't get as extreme as writing binary search (I have seen this in production C# code) or something using only bitshifting, feel free to leave this in.

Reply:
1 << zoom is basically 2^zoom which is a pretty slow operation, but I left a comment addressing the meaning of the bitshift

Reply:
Well you do have ` var bound = Math.pow(2, zoom) - normalizedCoord.y - 1;` in other places too. Maybe it secretly bitshifts behind the scene, haha.

Reply:
you're absolutely right. I'm not sure if its bitshifting precisely but the runtime seems to be the same